### PR TITLE
Improve File Manager Configuration Error Handling

### LINKED ISSUE #3217: BUG: Error when trying to open file with "Files" file explorer
### Checks

- [x] I have checked that this issue has not already been reported.

- [x] I am using the latest version of Flow Launcher.

### Problem Description

Basically when I switch my default file explorer to Files, and try to search for a folder and click open, this message popped up

[2025-01-31.txt](https://github.com/user-attachments/files/18618338/2025-01-31.txt)

### To Reproduce

1. change default file explorer to "files"
2. search any folder and click enter to open

### Screenshots

_No response_

### Flow Launcher Version

1.19.5

### Windows Build Number

Version 10.0.26100.3037

### Error Log

<details>

```shell


Flow Launcher version: 1.19.5
OS Version: 26100.3037
IntPtr Length: 8
x64: True

Python Path: C:\Users\xside\AppData\Roaming\FlowLauncher\Environments\Python\PythonEmbeddable-v3.11.4\pythonw.exe
Node Path: 
Date: 01/31/2025 12:31:25
Exception:
System.ComponentModel.Win32Exception (2): An error occurred trying to start process 'Files' with working directory 'C:\Users\xside\AppData\Local\FlowLauncher\app-1.19.5'. The system cannot find the file specified.
   at System.Diagnostics.Process.StartWithShellExecuteEx(ProcessStartInfo startInfo)
   at Flow.Launcher.PublicAPIInstance.OpenDirectory(String DirectoryPath, String FileNameOrFilePath) in C:\projects\flow-launcher\Flow.Launcher\PublicAPIInstance.cs:line 246
   at Flow.Launcher.Plugin.Explorer.Search.ResultManager.OpenFolder(String folderPath, String fileNameOrFilePath) in C:\projects\flow-launcher\Plugins\Flow.Launcher.Plugin.Explorer\Search\ResultManager.cs:line 346
   at Flow.Launcher.Plugin.Explorer.Search.ResultManager.<>c__DisplayClass14_0.<CreateOpenCurrentFolderResult>b__0(ActionContext c) in C:\projects\flow-launcher\Plugins\Flow.Launcher.Plugin.Explorer\Search\ResultManager.cs:line 263
   at Flow.Launcher.Plugin.Result.ExecuteAsync(ActionContext context) in /_/Flow.Launcher.Plugin/Result.cs:line 246
   at Flow.Launcher.ViewModel.MainViewModel.OpenResultAsync(String index) in C:\projects\flow-launcher\Flow.Launcher\ViewModel\MainViewModel.cs:line 389
   at CommunityToolkit.Mvvm.Input.AsyncRelayCommand.AwaitAndThrowIfFailed(Task executionTask)
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

```
</details>

### PR DESCRIPTION
**Pull request by onesounds**
### Summary
- Fix #3217, #3353, #3418
- When attempting to open a file using a file manager that no longer exists, a more informative message is shown instead of a generic error dialog.
- A warning message is now displayed if the specified file manager does not exist at the time of configuration.
- Updated the default path for Files.
- Added an informational note about Files in the Custom File Manager configuration window.

### Details

- The default executable path for **Files** has changed from `Files` to `Files-stable`. The default setting has been adjusted accordingly.
- Since the path may vary depending on the installed version, a note has been added to explain this. Additionally, warnings will be shown if the user tries to add or use an invalid path.
- Migration logic was considered, but due to complexity and edge cases, it was not implemented.

### Future

- In the future, it would be helpful to create a dedicated webpage where users can share configuration examples for various file managers.

### Test
- A warning popup should appear when attempting to register a file manager with a non-existent path.

- An error message should be shown when trying to open a folder using a file manager that was registered with a non-existent path.

### CODE REVIEW

File: Flow.Launcher/SelectFileManagerWindow.xaml.cs [hunk 1]
```diff
@@ -40,14 +45,119 @@
...
+        
+        private void Hyperlink_RequestNavigate(object sender, System.Windows.Navigation.RequestNavigateEventArgs e)
+        {
+            Process.Start(new ProcessStartInfo
+            {
+                FileName = e.Uri.AbsoluteUri,
+                UseShellExecute = true
+            });
+            e.Handled = true;
+        }
```
[the remark this answers is not in the document]
Comment (Jack251970) [reply Flow.Launcher/SelectFileManagerWindow.xaml.cs:57]: No need

File: Flow.Launcher/PublicAPIInstance.cs [hunk 2]
```diff
@@ -316,40 +316,63 @@ public void SavePluginSettings()
...
-                {
-                    FileName = explorerInfo.Path.Replace("%d", DirectoryPath),
-                    UseShellExecute = true,
-                    Arguments = FileNameOrFilePath is null
-                        ? explorerInfo.DirectoryArgument.Replace("%d", DirectoryPath)
-                        : explorerInfo.FileArgument
-                            .Replace("%d", DirectoryPath)
-                            .Replace("%f", targetPath)
-                };
+                // File Manager not found
```
[the remark this answers is not in the document]
Comment (Jack251970) [reply Flow.Launcher/PublicAPIInstance.cs]: Good!

File: Flow.Launcher/Languages/en.xaml [hunk 3]
```diff
@@ -366,13 +366,18 @@
...
     <system:String x:Key="fileManager_tips2">For example, if the file manager uses a command such as &quot;totalcmd.exe /A c:\windows&quot; to open the c:\windows directory, the File Manager Path will be totalcmd.exe, and the Arg For Folder will be /A &quot;%d&quot;. Certain file managers like QTTabBar may just require a path to be supplied, in this instance use &quot;%d&quot; as the File Manager Path and leave the rest of the fileds blank.</system:String>
     <system:String x:Key="fileManager_name">File Manager</system:String>
     <system:String x:Key="fileManager_profile_name">Profile Name</system:String>
     <system:String x:Key="fileManager_path">File Manager Path</system:String>
     <system:String x:Key="fileManager_directory_arg">Arg For Folder</system:String>
     <system:String x:Key="fileManager_file_arg">Arg For File</system:String>
+    <system:String x:Key="fileManagerPathNotFound">The path for the file manager '{0}' could not be found: '{1}' Do you want to continue?</system:String>
```
Comment (jjw24) [inline Flow.Launcher/Languages/en.xaml:379]:
```suggestion
    <system:String x:Key="fileManagerPathNotFound">The path for file manager '{0}' could not be found: '{1}' Do you want to continue?</system:String>
```

File: Flow.Launcher/SelectFileManagerWindow.xaml.cs [hunk 4]
```diff
@@ -40,14 +45,120 @@ private void btnCancel_Click(object sender, RoutedEventArgs e)
         {
             Close();
         }
+        
+        private void Hyperlink_RequestNavigate(object sender, System.Windows.Navigation.RequestNavigateEventArgs e)
```
Comment (jjw24) [inline Flow.Launcher/SelectFileManagerWindow.xaml.cs:49]: @Jack251970 do we need to put this code in the view?  Is there a viewmodel we can move this to?

File: Flow.Launcher/SelectFileManagerWindow.xaml.cs [hunk 5]
```diff
@@ -40,14 +45,120 @@ private void btnCancel_Click(object sender, RoutedEventArgs e)
...
+                FileName = e.Uri.AbsoluteUri,
+                UseShellExecute = true
+            });
+            e.Handled = true;
+        }
 
         private void btnDone_Click(object sender, RoutedEventArgs e)
         {
+            // Check if the selected file manager path is valid
+            if (!IsFileManagerValid(CustomExplorer.Path))
```
Comment (jjw24) [inline Flow.Launcher/SelectFileManagerWindow.xaml.cs:62]: Again can we move this logic in view model?

File: Flow.Launcher/SelectFileManagerWindow.xaml.cs [hunk 6]
```diff
@@ -40,14 +45,120 @@ private void btnCancel_Click(object sender, RoutedEventArgs e)
...
+                    return;
+                }
+            }
+
             _settings.CustomExplorerList = CustomExplorers.ToList();
             _settings.CustomExplorerIndex = SelectedCustomExplorerIndex;
             Close();
         }
 
+        private bool IsFileManagerValid(string path)
```
[the remark this answers is not in the document]
Comment (jjw24) [reply Flow.Launcher/SelectFileManagerWindow.xaml.cs:82]: FYI, MVVM - https://www.reddit.com/r/csharp/comments/yclkla/mvvm_where_to_put_the_ui_logic/

File: Flow.Launcher/SelectFileManagerWindow.xaml.cs (hunk 4 above)
Comment (Jack251970) [reply Flow.Launcher/SelectFileManagerWindow.xaml.cs:49]: I do not think this can be put into a VM

File: Flow.Launcher/SelectFileManagerWindow.xaml.cs (hunk 5 above)
Comment (Jack251970) [reply Flow.Launcher/SelectFileManagerWindow.xaml.cs]: Done

File: Flow.Launcher/SelectFileManagerWindow.xaml.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (Jack251970) [reply Flow.Launcher/SelectFileManagerWindow.xaml.cs]: Done

File: Flow.Launcher/ViewModel/SelectBrowserViewModel.cs [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        });
+        SelectedCustomBrowserIndex = CustomBrowsers.Count - 1;
+    }
+
+    [RelayCommand]
+    private void Delete()
+    {
+        CustomBrowsers.RemoveAt(SelectedCustomBrowserIndex--);
+    }
+}
```
[the remark this answers is not in the document]
Comment (Jack251970) [reply Flow.Launcher/ViewModel/SelectBrowserViewModel.cs:58]: Good!

File: Flow.Launcher/ViewModel/SelectFileManagerViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+            Name = "New Profile"
+        });
+        SelectedCustomExplorerIndex = CustomExplorers.Count - 1;
+    }
+
+    [RelayCommand]
+    private void Delete()
+    {
+        CustomExplorers.RemoveAt(SelectedCustomExplorerIndex--);
+    }
```
[the remark this answers is not in the document]
Comment (Jack251970) [reply Flow.Launcher/ViewModel/SelectFileManagerViewModel.cs]: Good!

File: Flow.Launcher/ViewModel/SelectFileManagerViewModel.cs [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+                    FileName = "where",
+                    Arguments = path,
+                    RedirectStandardOutput = true,
+                    UseShellExecute = false,
+                    CreateNoWindow = true
+                }
+            };
+            process.Start();
+            string output = process.StandardOutput.ReadToEnd();
+            process.WaitForExit();
```
[the remark this answers is not in the document]
Comment (Jack251970) [reply Flow.Launcher/ViewModel/SelectFileManagerViewModel.cs:91]: No need

**Review by jjw24 — APPROVED** [review]
